Jack/Property.fs

```diff
@@ -43,6 +43,13 @@ module Property =
         else
             Failure [] |> ofResult
 
+    let private convert (value : 'T) =
+        match box value with
+        | :? Lazy<bool> as b -> ofBool b.Value
+        | :? Property   as p -> p
+        | :? bool       as b -> ofBool b
+        | _                  -> ofBool true
```

Comment:
It looks like [some work was done](https://github.com/Microsoft/visualfsharp/compare/master...CaptainHayashi:hackathon-vs) on this... There's also [this](https://fslang.uservoice.com/forums/245727-f-language/suggestions/5762135-support-for-type-classes-or-implicits) UserVoice entry. Let's see what happens.